redfish-core/lib/assembly.hpp

```diff
@@ -365,9 +438,90 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+  
+        auto iter2 =
+            oemIndicatorMap.find(std::to_string(assemblyIndex));
+
+        if (iter2 != oemIndicatorMap.end())
+        {
+            std::optional<nlohmann::json> openbmc;
+            if (!json_util::readJson(iter2->second, asyncResp->res, "OpenBMC", openbmc))
+            {
+                BMCWEB_LOG_ERROR << "Property Value Format Error ";
+                messages::propertyValueFormatError(
+                                asyncResp->res, *openbmc,
+                                "OpenBMC");
+                return;
+            }
+
+            if (!openbmc)
+            {
+                BMCWEB_LOG_ERROR << "Property Missing ";
+                messages::propertyMissing(asyncResp->res, "OpenBMC");
+                return;
+            }
+
+            std::optional<bool> readytoremove;
+            if (!json_util::readJson(*openbmc, asyncResp->res,
+                                     "ReadyToRemove", readytoremove))
+            {
+                return;
+            }
+
+            if (!readytoremove)
+            {
+                BMCWEB_LOG_ERROR << "Property Missing ";
+                messages::propertyMissing(asyncResp->res, "ReadyToRemove");
+                return;
+            }
+
+            // Handle special case for tod_battery assembly OEM ReadyToRemove property.
+            // NOTE: The following method for the special case of the tod_battery
+            // ReadyToRemove property only works when there is only ONE adcsensor
+            // handled by the adcsensor application.
+            if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
```

Comment:
Should you have an else at the bottom with PropertyUnknown ? 
What if you PATCH ReadyToRemove on an assembly that isn't the tod_battery ?